blockbase/inc/customizer/wp-customize-fonts.php

```diff
@@ -504,51 +497,41 @@ function handle_customize_save_after( $wp_customize ) {
 
-		$user_theme_json_post         = get_post( $user_custom_post_type_id );
-		$user_theme_json_post_content = json_decode( $user_theme_json_post->post_content );
+		// API request to get global styles
+		$get_request = new WP_REST_Request( 'GET', '/wp/v2/global-styles/' );
+		$get_request->set_param( 'id', $user_custom_post_type_id );
 
-		// Set meta settings.
-		$user_theme_json_post_content->version                     = 1;
-		$user_theme_json_post_content->isGlobalStylesUserThemeJSON = true;
+		$global_styles_controller = new Gutenberg_REST_Global_Styles_Controller();
+		$global_styles            = $global_styles_controller->get_item( $get_request );
 
-		$this->unset_property_if_it_exists( $user_theme_json_post_content->settings->typography, 'fontFamilies' );
+		// converts data to array (in some cases settings and styles are objects insted of arrays)
+		$new_settings =  (array) $global_styles->data['settings'];
+		$new_styles   =  (array) $global_styles->data['styles'];
 
-		// Set the typography settings.
-		$user_theme_json_post_content = set_settings_array(
-			$user_theme_json_post_content,
-			array( 'settings', 'typography', 'fontFamilies', 'custom' ),
-			$font_families
-		);
+		// Set new typography settings
+        if ( $font_families ) {
+            $new_settings['typography']['fontFamilies']['custom'] = $font_families;
+        }
 
-		//If the typeface choices === the default then we remove it instead
+		// Removes typography settings if the default are selected
 		if ( $body_value === $body_default && $heading_value === $heading_default ) {
-			// These lines need to stay for backwards compatibility.
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->typography, 'fontFamily' );
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->elements->h1->typography, 'fontFamily' );
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->elements->h2->typography, 'fontFamily' );
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->elements->h3->typography, 'fontFamily' );
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->elements->h4->typography, 'fontFamily' );
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->elements->h5->typography, 'fontFamily' );
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->elements->h6->typography, 'fontFamily' );
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->blocks->{'core/button'}->typography, 'fontFamily' );
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->blocks->{'core/post-title'}->typography, 'fontFamily' );
-			$this->unset_property_if_it_exists( $user_theme_json_post_content->styles->blocks->{'core/pullquote'}->typography, 'fontFamily' );
+			unset( $new_settings['typography'] );
```

Comment:
I think this should probably only unset $new_settings['typography']['fontFamilies']['custom'], since the user might make other customizations using Global Styles.

Reply:
Doing so makes the customizer behave poorly.  (I wasn't able to get to the bottom of why.  Though I agree with the sentiment @scruffian.

Much like the "remove customizations when settings are default" was causing more problems then they were solving for colors I think the same is true here.  I argue that the whole if block should be removed.  Testing of that shows things continuing to work as expected without the possible side effect of removing customizations made elsewhere.

Reply:
I removed the `if` block.